src/newrelic/utilization/utilization_hash.go

```diff
@@ -204,6 +199,38 @@ func GatherDockerID(util *Data) error {
 	return nil
 }
 
+func OverrideDockerId(util *Data, id string) {
+	if nil == util.Vendors {
+		util.Vendors = &vendors{}
+		util.Vendors.Docker = &docker{}
+	}
+	util.Vendors.Docker = &docker{ID: id}		
+}
+
+func OverrideVendors(util *Data) {
+	if util.Vendors.isEmpty() {
```

Comment:
Should this have checks for util != nil and util.Vendors != nil?

Reply:
I believe the logic to check if `Vendors == nil` is handled implicitly by the call to the isEmpty function here: https://github.com/newrelic/newrelic-php-agent/blob/main/src/newrelic/utilization/utilization_hash.go#L70

Reply:
If 'util' or `util.Vendors` is nil how can one call `util.Vendors.isEmpty()` w/o an error?

Reply:
Ok it makes sense that in golang you can run the functions for a data type on a data type value of `nil` so no changes are needed to check for `utils.Vendors`.  
However, if `util` is `nil` I think it could cause a problem since accessing `util.Vendors` will cause an error.

Reply:
See how you do it below in `GetDockerId()` where you return early if `util` is `nil`.